torch/nn/modules/adaptive.py

```diff
@@ -167,34 +167,54 @@ def reset_parameters(self) -> None:
             h2o.reset_parameters()
 
     def forward(self, input: Tensor, target: Tensor) -> _ASMoutput:
-        if input.size(0) != target.size(0):
-            raise RuntimeError('Input and target should have the same size '
-                               'in the batch dimension.')
+        targ_dim = target.dim()
+
+        if targ_dim == 1:
+            if input.size(0) != target.size(0):
+                raise RuntimeError('Input and target should have the same size '
+                                   'in the batch dimension.')
+            if input.dim() != 2:
+                raise RuntimeError('1D target tensor expects 2D input tensors, ' 
+                                   'but found inputs with size', input.size())
+        elif targ_dim == 0:
+            if input.dim() != 1:
+                raise RuntimeError('0D target tensor expects 1D input tensors, ' 
+                                   'but found inputs with size', input.size())
+        else:
+            raise RuntimeError('0D or 1D target tensor expected, ' 
+                               'multi-target not supported')
+
+        if targ_dim == 0:
+            new_input = input.unsqueeze(0)
+            new_target = target.unsqueeze(0)
+        else:
+            new_input = input
+            new_target = target
```

Comment:
Dang, they duplicated this logic across python and C++? I thought only MultiHeadAttention was like that :p

Similar comments as above then apply here as well